### PR TITLE
Dog found value

### PR DESCRIPTION
**Pull request by itamar-cord**

### CODE REVIEW

File: app/routers/dogfinder.py
Comment (YanDav78) [inline app/routers/dogfinder.py:1]: Please keep the same formatting as before. It will be easier to review the code changes.

File: app/routers/dogfinder.py [hunk 1]
```diff
@@ -25,48 +25,53 @@
...
+    ]
+}
 
 
 # This is the request model for the add_documents endpoint
 class DocumentRequest(BaseModel):
     documents: List[DogDocument]
 
+
+is_found_field = "isFound"
```
Comment (YanDav78) [inline app/routers/dogfinder.py]: This should be uppercase as it is constant. 
It is better to add it on the DogDocument model and the schema itself (dog_class_definition).

File: app/routers/dogfinder.py [hunk 2]
```diff
@@ -25,48 +25,53 @@
...
+uuid_field = "uuid5"
```
Comment (YanDav78) [inline app/routers/dogfinder.py]: We don't need to add another field name uuid5. In order to get the id you need to ask it in .with_additional(["distance"]) and change it to .with_additional(["distance", "id"]).
Please revert this change

File: app/routers/dogfinder.py [hunk 3]
```diff
@@ -76,14 +81,16 @@ async def startup_event():
...
+    vecotrDBClient = WeaviateVectorDBClient(
+        url=f"http://{os.getenv('WEAVIATE_HOST', 'localhost')}:{os.getenv('WEAVIATE_PORT', '8080')}")
+
     # Create the schema
     vecotrDBClient.create_schema(class_name="Dog", class_obj=dog_class_definition)
 
 
 @router.post("/query/", response_model=APIResponse)
-async def query(query: Optional[str] = Form(None), type: str = Form(...), breed: Optional[str] = Form(None), img: UploadFile = File(...), top: int = Form(10)):
+async def query(query: Optional[str] = Form(None), type: str = Form(...), breed: Optional[str] = Form(None),
```
Comment (YanDav78) [inline app/routers/dogfinder.py]: Same here with the formatting

File: app/routers/dogfinder.py [hunk 4]
```diff
@@ -76,14 +81,16 @@ async def startup_event():
     global vecotrDBClient
 
     # Create the vector db client, connecting to the weaviate instance
-    vecotrDBClient = WeaviateVectorDBClient(url=f"http://{os.getenv('WEAVIATE_HOST', 'localhost')}:{os.getenv('WEAVIATE_PORT', '8080')}")
-    
+    vecotrDBClient = WeaviateVectorDBClient(
```
Comment (YanDav78) [inline app/routers/dogfinder.py]: formatting

File: app/routers/dogfinder.py [hunk 5]
```diff
@@ -142,8 +154,9 @@ async def add_document(type: str = Form(...), breed: Optional[str] = Form(None),
             # Create the data object
             data_properties = create_data_properties(dogDocument)
             # Create a uuid based on the filename
-            data_properties["uuid5"] = generate_uuid5({ "filename": dogDocument.filename })
+            data_properties[uuid_field] = generate_uuid5({"filename": dogDocument.filename})
```
Comment (YanDav78) [inline app/routers/dogfinder.py]: No need. See comment above, regarding uuid5

File: app/routers/dogfinder.py [hunk 6]
```diff
@@ -174,26 +188,30 @@ async def add_documents(documentRequest: DocumentRequest):
         # Create the embedding model
         logger.info(f"Creating embedding model")
         embedding_model, cache_info = create_embedding_model()
-            
+
         # Embed the documents images
-        embedding_results = embed_documents([create_pil_image(document.image) for document in documentRequest.documents], embedding_model)
+        embedding_results = embed_documents(
```
Comment (YanDav78) [inline app/routers/dogfinder.py]: formatting

File: app/routers/dogfinder.py [hunk 7]
```diff
@@ -174,26 +188,30 @@ async def add_documents(documentRequest: DocumentRequest):
...
         # Iterate over the documents and add them to the database
         for i, document in enumerate(documentRequest.documents):
             try:
                 # Create the data object
-                data_properties = create_data_properties(document)       
-         
+                data_properties = create_data_properties(document)
+
                 # Create a uuid based on the filename
-                data_properties["uuid5"] = generate_uuid5({ "filename": document.filename })
```
Comment (YanDav78) [inline app/routers/dogfinder.py:188]: Same comment as above with the uuid5. no need for this.

File: app/services/weaviate_vectordb_client.py [hunk 8]
```diff
@@ -52,7 +53,7 @@ def check_batch_result(results: dict):
             # Iterate over the documents and add them to the database
             for i, data_properties in enumerate(documents):
                 try:
-                    uuid5 = data_properties.pop("uuid5")
```
Comment (YanDav78) [inline app/services/weaviate_vectordb_client.py]: No need for this as well. Please revert

File: app/services/weaviate_vectordb_client.py [hunk 9]
```diff
@@ -125,7 +126,8 @@ def query(self, class_name: str, query: str, query_embedding: List[float], limit
             logger.error(f"Error while querying the database: {results['errors']}")
             # raise exception, not http
             raise Exception(f"Error while querying the database: {results['errors']}")
-
```
Comment (YanDav78) [inline app/services/weaviate_vectordb_client.py]: Formatting

File: app/routers/dogfinder.py
Comment (itamar-cord) [reply app/routers/dogfinder.py:1]: ok

File: app/routers/dogfinder.py (hunk 3 above)
Comment (itamar-cord) [reply app/routers/dogfinder.py]: ok

File: app/routers/dogfinder.py (hunk 4 above)
Comment (itamar-cord) [reply app/routers/dogfinder.py]: done

File: app/routers/dogfinder.py (hunk 5 above)
Comment (itamar-cord) [reply app/routers/dogfinder.py]: This field is still used to put the id into the db,not just a pull thing, if I remove it I the db wont have an id to use

File: app/routers/dogfinder.py (hunk 6 above)
Comment (itamar-cord) [reply app/routers/dogfinder.py]: done

File: app/routers/dogfinder.py (hunk 1 above)
Comment (itamar-cord) [reply app/routers/dogfinder.py]: moved to upper case, used in the schema

File: app/routers/dogfinder.py (hunk 2 above)
Comment (itamar-cord) [reply app/routers/dogfinder.py]: used the with_additonal, but the field is still used in multiple spots so I kept it

File: app/routers/dogfinder.py (hunk 7 above)
Comment (itamar-cord) [reply app/routers/dogfinder.py:188]: same response, it is used in the insert

File: app/services/weaviate_vectordb_client.py (hunk 8 above)
Comment (itamar-cord) [reply app/services/weaviate_vectordb_client.py]: reverted

File: app/services/weaviate_vectordb_client.py (hunk 9 above)
Comment (itamar-cord) [reply app/services/weaviate_vectordb_client.py]: done

File: app/routers/dogfinder.py [hunk 10]
```diff
@@ -142,8 +150,9 @@ async def add_document(type: str = Form(...), breed: Optional[str] = Form(None),
             # Create the data object
             data_properties = create_data_properties(dogDocument)
             # Create a uuid based on the filename
-            data_properties["uuid5"] = generate_uuid5({ "filename": dogDocument.filename })
+            data_properties[UUID_FIELD] = generate_uuid5({"filename": dogDocument.filename})
             data_properties["document_embedding"] = dog_embedding
+            data_properties[IS_FOUND_FIELD] = False
```
Comment (YanDav78) [inline app/routers/dogfinder.py]: You can just put a default value on the DogDocument to False and you don't need this line

File: app/routers/dogfinder.py [hunk 11]
```diff
@@ -183,14 +192,16 @@ async def add_documents(documentRequest: DocumentRequest):
...
+
                 # Create a uuid based on the filename
-                data_properties["uuid5"] = generate_uuid5({ "filename": document.filename })
+                data_properties[UUID_FIELD] = generate_uuid5({"filename": document.filename})
 
                 # Embed the document
                 logger.info(f"Setting Embedding document [{document.filename}] {i+1} of {documents_length}")
                 data_properties["document_embedding"] = embedding_results[i]
 
+                data_properties[IS_FOUND_FIELD] = False
```
Comment (YanDav78) [inline app/routers/dogfinder.py]: You can just put a default value on the DogDocument to False and you don't need this line

**Review by YanDav78 — APPROVED** [review]
